lib/interceptaccess.c

```diff
@@ -37,10 +37,17 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <sys/mman.h>
 
 #include "libpreopen.h"
 
-int access(const char *pathname,int mode)
+int access(const char *path, int mode)
 {
-	return faccessat(AT_FDCWD,pathname,mode, AT_EACCESS);
+		
+	struct po_map *map = po_map_get();
+
+	struct po_relpath rel = po_find(map, path, NULL);
+
+
+	return faccessat(rel.dirfd, rel.relative_path, mode,AT_EACCESS );
```

Comment:
I think we want a flag value of 0 rather than `AT_EACCESS`: this function should behave as closely to `access(2)` as possible, but `AT_EACCESS` changes the semantics.